src/runtime/internal/Component.ts

```diff
@@ -52,23 +53,26 @@ export function claim_component(block, parent_nodes) {
 	block && block.l(parent_nodes);
 }
 
-export function mount_component(component, target, anchor) {
+export function mount_component(component, target, anchor, customElement) {
 	const { fragment, on_mount, on_destroy, after_update } = component.$$;
 
 	fragment && fragment.m(target, anchor);
 
-	// onMount happens before the initial afterUpdate
-	add_render_callback(() => {
-		const new_on_destroy = on_mount.map(run).filter(is_function);
-		if (on_destroy) {
-			on_destroy.push(...new_on_destroy);
-		} else {
-			// Edge case - component was destroyed immediately,
-			// most likely as a result of a binding initialising
-			run_all(new_on_destroy);
-		}
-		component.$$.on_mount = [];
-	});
+	if (!customElement) {
```

Comment:
I can't see an obvious way to do it, but I'm wondering if it's possible/worthwhile making the alternate functionality tree-shakable so it's only included in the build if you're creating non-custom elements?

Reply:
Good idea! I'm not sure how to achieve that either though...

Reply:
I guess the same could be done for the on_disconnect callback array.

Reply:
Seems like a fun idea, but I really hope this PR wouldn't block on it being released. The number of bytes saved would be pretty tiny, and this PR fixes a fairly crucial bug.

Reply:
Agreed. Certainly something that could be optimised later.